test/suites/host-ip-change.robot

```diff
@@ -0,0 +1,233 @@
+
+Rollback Configuration
+    [Arguments]    ${backup_file}
+    [Documentation]    Restore the original configuration from the backup file
+    Restore Backup File    ${backup_file}
+
+Add Node To Hosts File
+    [Arguments]    ${ip}    ${node_name}
+    [Documentation]    Add the node name and IP address to the /etc/hosts file
+    ${stdout}    ${rc}=    Execute Command
+    ...    echo "${ip} ${node_name}" | sudo tee -a /etc/hosts
+    ...    sudo=True    return_rc=True
+    Should Be Equal As Integers    ${rc}    0
+
+Get NodeName
+    [Documentation]    Get and Return nodename
+    ${nodeName}=    Run With Kubeconfig    oc get node --no-headers | awk '{print $1}'
+    Should Not Be Empty    ${nodeName}
+    [RETURN]    ${nodeName}
+
+Modify NetworkManager Configuration
+    [Arguments]    ${ip}    ${gateway}   ${device}
+    [Documentation]    Modify NetworkManager configuration using nmcli commands
+    ${stdout}   ${rc1}=    Execute Command    nmcli con modify ${device} ipv4.addresses ${ip}
+    ...    sudo=True    return_rc=True
+    ${stdout}   ${rc2}=    Execute Command    nmcli con mod ${device} ipv4.gateway ${gateway}
+    ...    sudo=True    return_rc=True
+    ${stdout}   ${rc3}=    Execute Command    nmcli con mod ${device} ipv4.method manual
+    ...    sudo=True    return_rc=True
+    Should Be Equal As Integers    ${rc1}     0
+    Should Be Equal As Integers    ${rc2}     0
+    Should Be Equal As Integers    ${rc3}     0
+
+Login MicroShift Host With New IP
+  [Arguments]  ${ip}
+  [Documentation]  Login to the MicroShift host via ssh and leave the connection open
+  ...  This keyword is meant to be used at the suite level. This ensures
+  ...  most tests already have an open connection. Any tests that will take
+  ...  action that disrupt that connection are responsible for restoring it.
+
+  Log  Host: ${ip}
+  IF    '${SSH_PORT}'
+        SSHLibrary.Open Connection    ${ip}    port=${SSH_PORT}
+  ELSE
+        SSHLibrary.Open Connection    ${ip}
+  END
+  Wait And Retry Login MicroShift Host  ${ip}
+  # If there is an ssh key set in the global configuration, use that to
+  # login to the host. Otherwise, assume that the ssh agent is running
+  # and configured properly.
+  IF  '${SSH_PRIV_KEY}'
+    SSHLibrary.Login With Public Key  ${USHIFT_USER}  ${SSH_PRIV_KEY}
+  ELSE
+    SSHLibrary.Login  ${USHIFT_USER}  allow_agent=True
+  END
+
+Wait And Retry Login MicroShift Host
+  [Arguments]  ${ip}
+  [Documentation]  Wait and retry logging in to the MicroShift host via SSH
+   ...  with a specified maximum number of retries and interval between retries.
+  Wait Until Keyword Succeeds  20x  15s  SSHLibrary.Login With Public Key  ${USHIFT_USER}  ${SSH_PRIV_KEY}
+
+Reboot Microshift System
+    [Documentation]    Reboot the system using "systemctl reboot" command
+    ${stdout}    ${rc}=    Execute Command    systemctl reboot
+    ...    sudo=True    return_rc=True
+    Log    Reboot command executed. Return code: ${rc}
```

Comment:
Return values from keywords should be logged automatically.

Reply:
Done